cpp/src/slice2php/Main.cpp

```diff
@@ -810,9 +810,8 @@ CodeVisitor::visitDictionary(const DictionaryPtr& p)
     }
     else if (!dynamic_pointer_cast<Enum>(keyType))
     {
-        // TODO: using 'InvalidMetadata' as our warning category for an unsupported key type feels weird.
         // See https://github.com/zeroc-ice/ice/issues/254
-        p->unit()->warning(p->file(), p->line(), InvalidMetadata, "dictionary key type not supported in PHP");
```

Comment:
`All` means that this warning doesn't have a category, since in order to suppress it, you have to suppress `All` warnings.

This feels more correct than lumping this in with 'invalid metadata'.